src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+
+  convertOfs.write(reinterpret_cast<const char *>(&minX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&minY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&index), sizeof(uint64_t));
+}
+
+void Rtree::SaveEntryWithOrderIndex(RTreeValueWithOrderIndex treeValue, std::ofstream& convertOfs) {
+  /**
+     * Save a single entry, containing its postion in the x- and y-sorting
+   */
+  double minX = treeValue.box.min_corner().get<0>();
+  double minY = treeValue.box.min_corner().get<1>();
+  double maxX = treeValue.box.max_corner().get<0>();
+  double maxY = treeValue.box.max_corner().get<1>();
+
+  convertOfs.write(reinterpret_cast<const char *>(&minX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&minY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxX), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&maxY), sizeof(double));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.id), sizeof(uint64_t));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.orderX), sizeof(uint64_t));
+  convertOfs.write(reinterpret_cast<const char *>(&treeValue.orderY), sizeof(uint64_t));
+}
+
+multiBoxGeo Rtree::LoadEntries(const std::string& file) {
+  multiBoxGeo boxes;
+
+  FileReaderWithoutIndex fileReader = FileReaderWithoutIndex(file);
+
+  std::optional<RTreeValue> element = fileReader.GetNextElement();
+  while (element) {
+    boxes.push_back(element.value());
+    element = fileReader.GetNextElement();
+  }
+
+  fileReader.Close();
+
+  return boxes;
+}
+
+multiBoxWithOrderIndex Rtree::LoadEntriesWithOrderIndex(const std::string& file) {
+  multiBoxWithOrderIndex boxes;
+  FileReader fileReader = FileReader(file);
+
+  std::optional<RTreeValueWithOrderIndex> element = fileReader.GetNextElement();
+  while (element) {
+    boxes.push_back(element.value());
+    element = fileReader.GetNextElement();
+  }
+
+  fileReader.Close();
+
+  return boxes;
+}
+
+bool OrderedBoxes::WorkInRam() const{
+  return this->workInRam;
+}
+
+void OrderedBoxes::CreateOrderedBoxesInRam(RectanglesForOrderedBoxes& rectanglesD0, RectanglesForOrderedBoxes& rectanglesD1, Rtree::BoundingBox box) {
+  this->workInRam = true;
+  this->rectsD0 = rectanglesD0;
+  this->rectsD1 = rectanglesD1;
+  this->size = (*rectsD0.rectanglesInRam).size();
+  this->boundingBox = box;
+}
+
+void OrderedBoxes::CreateOrderedBoxesOnDisk(RectanglesForOrderedBoxes& rectanglesD0, RectanglesForOrderedBoxes& rectanglesD1, uint64_t size, Rtree::BoundingBox box) {
+  this->workInRam = false;
+  this->rectsD0 = rectanglesD0;
+  this->rectsD1 = rectanglesD1;
+  this->size = size;
+  this->boundingBox = box;
+}
+
+Rtree::BoundingBox OrderedBoxes::GetBoundingBox() {
+  return this->boundingBox;
```

Comment:
The `this->` is not necessary (for all of those functions).
Especially once your members all end with an `underscore_`, it is even visible, that they are members.